reVX/least_cost_xmission/README.md

```diff
@@ -1,25 +1,145 @@
+
+* *Costs* - The cost of building a transmission line including ROW and any factors such as slope, land use, and water depth though a single grid cell (typically 90 meters.) The routing algorithm automatically scales costs for any transmission crossing a cell at a diagonal. Costs for land-based transmission and undersea cables are calculated separately and then combined. Note that landfall costs are incorporated into the final cost raster when wet and dry costs are combined.
+* *Friction* - Friction indicates areas that are undesirable for transmission. Paths that must cross a friction area will take the shortest possible path. Friction does not affect the cost of a line that runs across it; it merely discourages a path from crossing a friction area, possibly taking a longer and more expensive path.
+* *Barriers* - Barriers are effectively a much stronger deterrent for transmission than friction. Routing will preferentially take a more expensive, and higher friction route before crossing a barrier area. Barriers do not affect the cost of lines that are forced to cross them.
+
+Note that the *friction* and *barriers* layers must be combined together, using a multiplier for barriers, before an analysis is ran.
+
+Layers are created by passing a JSON configuration file (config file) to the [`transmission-layer-creator`](transmission_layer_creator_cli.py) command-line tool. The format of the JSON file is defined using [Pydantic](https://docs.pydantic.dev/latest/) in the `LayerCreationConfig` class of [`transmission_layer_creation.py`](../config/transmission_layer_creation.py). The config file consists of key-value pairs describing necessary files and layer creation options and can trigger a number of different operations depending on it's contents. These operations include:
+
+* Creating a new H5 file to store layers in
+* Creating wet, dry, and landfall masks from a vector
+* Creating wet and dry (TODO) cost layers
+* Combining wet and dry costs
+* Creating frictions and/or barrier layers
+* Combining friction and barrier layers together
+
+## Config File Keys
+### Required Keys
+The following keys are required to run any type of layer creation:
+
+* `template_raster_fpath` - Template GeoTIFF with the shape, CRS, and transform to use for the project. Data in the raster is ignored.
+* `h5_fpath` - H5 file to store layers in. `h5_fpath` will be created if it does not exist and `existing_h5_fpath` is set.
+
+### Optional Keys
+Most of these keys are optional and affect how the layer creation runs. Some, like `land_mask_vector_fname`, may need to be specified once to generate the mask files, and then can be disabled.
+
+* `existing_h5_fpath` - Existing reV H5 file with the same shape as `template_raster_fpath`. Metadata is copied from `existing_h5_fpath` to `h5_fpath`. This must be defined to create `h5_fpath` if it does not exist.
```

Comment:
Oh, yea it seems like we should merge the `existing_h5_fpath` and `template_raster_fpath` into a single input argument (i.e. `template_fpath`)? And it should accept wither a tiff or h5 file. I think you should be able to pull all relevant info from just one of these inputs, right?

Reply:
Yes, in theory. I'm not fully sure what is supposed to be in the global attrs for an H5. On eagle, /projects/rev/data/conus/rasters/exclusions/Offshore_Exclusions.h5 only has `creation_date`, `profile`, and `shape`, though the profile is also specified on each layer, and the shape could arguably be different for each layer as well, though I'm guessing the reV convention is they all share the same shape.

I would have to lookup how to generate the lats and longs from transform and CRS, but that shouldn't be hard.

Reply:
Yea the reV convention is that all layers are the same shape. If that doesn't hold, quite a few things break.

I'm not sure if there is a formal convention for what is required in the attrs, but [this ExclusionsCoinverter init code](https://github.com/NREL/reVX/blob/564d8aeeec95cd850aa65c46e8f950771d36ff10/reVX/utilities/exclusions.py#L628-L672) is what we should be running when creating those files, so I would go with what it implements. That code also shows how to pull lat/lons from a Tiff (the Geotiff class does that for you), so I would just copy that. As I mentioned before, I'll probably merge some of this functionality down the line, since we are duplicating efforts, but it's fine for now.